lib/rubocop/cop/rspec/stubbed_mock.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that message expectations do not have a configured response.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to receive(:bar).with(42).and_return("hello world")
+      #
+      #   # good (without spies)
+      #   allow(foo).to receive(:bar).with(42).and_return("hello world")
+      #   expect(foo).to receive(:bar).with(42)
+      #
+      class StubbedMock < Base
+        MSG = 'Prefer `%<replacement>s` to `%<method_name>s` when ' \
+              'configuring a response.'
+
+        def_node_matcher :message_expectation?, <<-PATTERN
+          {
+            (send nil? { :receive :receive_message_chain } ...)
```

Comment:
In some cops, we also match `:stub_chain`. Should we do the same here? Or, is it soon time to remove compatibility with the deprecated RSpec syntax?

Reply:
You are quite right that this cop is getting really hard to grasp. I'm inclined not to add support for a rather obscure and deprecated syntax, further complicating it.

If that wouldn't be complicating the cop, I'm all for supporting the deprecated syntax, even when RSpec 4 is out.
Users might still stick with RSpec 3, and I thought we may still support it for some period of time, just like [`rspec-rails` supports Rails 4.2](https://github.com/rubocop-hq/rubocop-rails/blob/159eef021279532312af9b7b89331bd44a7eeb9d/spec/support/shared_contexts.rb#L4).